amivapi/tests/users/test_user_cleanup.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+#
+# license: AGPLv3, see LICENSE for details. In addition we strongly encourage
+#          you to buy us beer if we meet and you like the software.
+"""Tests for user module.
+
+Test if inactive non-members get deleted from database
+"""
+
+from freezegun import freeze_time
+
+from amivapi.tests import utils
+from amivapi.users import cleanup
+
+
+class UserCleanupTest(utils.WebTest):
+    """Test automatic User Cleanup"""
+
+    def test_autodelete(self):
+        entries = [
+            {
+                'nethz': 'pablamiv',
+                'firstname': "Pabla",
+                'lastname': "AMIV",
+                'email': "pabla@amiv.ch",
+                'gender': 'male',
+                'membership': 'none'
+            },
+            {
+                'nethz': 'pablomiv',
+                'firstname': "Pablo",
+                'lastname': "AMIV",
+                'email': "pablo@amiv.ch",
+                'gender': 'male',
+                'membership': "regular"
+            },
+            {
+                'nethz': 'pablemiv',
+                'firstname': "Pable",
+                'lastname': "AMIV",
+                'email': "pable@amiv.ch",
+                'gender': 'male',
+                'membership': "none"
+            },
+            {
+                'nethz': 'pablimiv',
+                'firstname': "Pabli",
+                'lastname': "AMIV",
+                'email': "pabli@amiv.ch",
+                'gender': 'male',
+                'membership': "honorary"
+            }
+        ]
+
+        root_token = self.get_root_token()
+
+        with self.app.app_context():
+            with freeze_time("2019-08-24T14:15:22Z"):
+                # Non-member, updated 6 Month ago
+                self.api.post("/users", data=entries[0], token=root_token)
+                # Member, updated 6 Month ago
+                self.api.post("/users", data=entries[1], token=root_token)
+
+            with freeze_time("2018-08-24T14:15:22Z"):
+                # Non-member, updated 18 Month ago (should be deleted)
+                self.api.post("/users", data=entries[2], token=root_token)
+                # (honorary) Member, updated 18 Month ago
+                self.api.post("/users", data=entries[3], token=root_token)
```

Comment:
`updated 6 Month ago` is a bit confusing because the time was fixed to a specific time instant which is more than 6 months ago.

I would suggest to rewrite this part of the code as follows (linked to the suggested changes above):

```suggestion
            for idx, entry in enumerate(entries):
                with freeze_time(entry.get('updated')):
                    user = {
                        'nethz': f'pabloamiv{str(idx)}',
                        'firstname': 'Pablo',
                        'lastname': "AMIV",
                        'email': f'pablo{str(idx)}@amiv.ch',
                        'gender': 'male',
                        'membership': entry.get('membership')
                    }
                    self.api.post("/users", data=user, token=root_token)
```